Added Gazebo, Navigation2, MoveIt, and micro-ROS to related projects (#2323)
ros2/ros2_documentation: source/Related-Projects.rst

Related Projects
================

Ignition Gazebo
---------------

**Gazebo** and its successor **Ignition Gazebo** `(gazebosim.org) <http://gazebosim.org/>`_ are the first open-source choice for 3D physics simulation of ROS-based robots.

Large Community Projects
------------------------

Large community projects involve multiple developers from all over the globe and are typically backed by a dedicated working group (cf. :ref:`governance`).

* **Navigation2** `(navigation.ros.org) <https://navigation.ros.org/>`_: Comprehensive and flexible navigation stack for mobile robots using ROS 2.
* **MoveIt** `(moveit.ros.org) <https://moveit.ros.org/>`_: A rich platform for building manipulation applications featuring advanced kinematics, motion planning, control, collision checking, and much more.
* **micro-ROS** `(micro.ros.org) <https://micro.ros.org/>`_ puts ROS 2 onto microcontrollers, starting at less than 100 kB of RAM.

Further Community Projects
--------------------------

> The global ROS community develops maintains hundreds of further packages on top of the core ROS 2 stack. Some of them come with own websites for documentation. Your best entry point to discover these works is the **ROS Index** `(index.ros.org) <https://index.ros.org/packages/>`_.

Reviewer: ```suggestion
The global ROS community develops maintains hundreds of further packages on top of the core ROS 2 stack. Some of them come with their own websites for documentation. Your best entry point to discover these works is the **ROS Index** `(index.ros.org) <https://index.ros.org/packages/>`_.
```

> *Hint for developers:* If you maintain a ``README.md`` file in the root of your package folder (which is not necessary the root of your repository), this file is rendered into the overview page of your package at `index.ros.org <https://index.ros.org/packages/>`_. The file may be used for documenting your package and supersedes the package documentation pages in the ROS Wiki from ROS 1. See the `fmi_adapter package <https://index.ros.org/p/fmi_adapter/github-boschresearch-fmi_adapter/#foxy>`_ as an example.

Reviewer: ```suggestion
*Hint for developers:* If you maintain a ``README.md`` file in the root of your package folder (which is not necessarily the root of your repository), this file is rendered into the overview page of your package at `index.ros.org <https://index.ros.org/packages/>`_. The file may be used for documenting your package and supersedes the package documentation pages in the ROS Wiki from ROS 1. See the `fmi_adapter package <https://index.ros.org/p/fmi_adapter/github-boschresearch-fmi_adapter/#foxy>`_ as an example.
```

Reviewer: Can you also maybe start sentences from newlines? It will make it easier to review

Related Projects
================

Ignition Gazebo
---------------

**Gazebo** and its successor **Ignition Gazebo** `(gazebosim.org) <http://gazebosim.org/>`_ are the first open-source choice for 3D physics simulation of ROS-based robots.

Large Community Projects
------------------------

> Large community projects involve multiple developers from all over the globe and are typically backed by a dedicated working group (cf. :ref:`governance`).

Reviewer 2: We prefer to use the `:doc:` tag for internal links:

```suggestion
Large community projects involve multiple developers from all over the globe and are typically backed by a dedicated working group (cf. :doc:`Governance`).
```

* **Navigation2** `(navigation.ros.org) <https://navigation.ros.org/>`_: Comprehensive and flexible navigation stack for mobile robots using ROS 2.
* **MoveIt** `(moveit.ros.org) <https://moveit.ros.org/>`_: A rich platform for building manipulation applications featuring advanced kinematics, motion planning, control, collision checking, and much more.

> * **micro-ROS** `(micro.ros.org) <https://micro.ros.org/>`_ puts ROS 2 onto microcontrollers, starting at less than 100 kB of RAM.

Reviewer 2: This bullet item is in a different "style" than the previous two.  I'll suggest:

```suggestion
* **micro-ROS** `(micro.ros.org) <https://micro.ros.org/>`_: A platform for putting ROS 2 onto microcontrollers, starting at less than 100 kB of RAM.*
```

Author: Is the asterisk at the end required?

Reviewer 2: > Is the asterisk at the end required?

Ah, no, that was a typo.  You did the right thing in your update.

Author: Changed as proposed in https://github.com/ros2/ros2_documentation/pull/2323/commits/46d51283161f4f4da4329b2b8c1457f89107eed2.

Further Community Projects
--------------------------

> The global ROS community develops maintains hundreds of further packages on top of the core ROS 2 stack.

Reviewer: ```suggestion
The global ROS community develops and maintains hundreds of further packages on top of the core ROS 2 stack.
```
